pkg/controller/registry/reconciler/configmap.go

```diff
@@ -214,28 +214,36 @@ func (c *ConfigMapRegistryReconciler) currentRoleBinding(source configMapCatalog
 
 func (c *ConfigMapRegistryReconciler) currentPods(source configMapCatalogSourceDecorator, image string) []*v1.Pod {
 	podName := source.Pod(image).GetName()
-	pods, err := c.Lister.CoreV1().PodLister().Pods(source.GetNamespace()).List(labels.SelectorFromSet(source.Selector()))
+	pods, err := c.OpClient.KubernetesInterface().CoreV1().Pods(source.GetNamespace()).List(context.TODO(), metav1.ListOptions{LabelSelector: labels.SelectorFromSet(source.Selector()).String()})
```

Comment:
This may technically solve the issue, but if at all possible, talking directly to the kube api for `read` requests should be avoided.

The operator is already watching pods, so it will eventually know about all of these pods, and this just adds load to the apiserver.

Reply:
Thanks for comments.  OK.  The second pod stays up unused until the next change of the catalogsource.  I can probably change code to stop the second pod sooner without issuing the read requests instead preventing the second pod creation.